ref/api-reference.md

```diff
@@ -1,7 +1,174 @@
+
+The codes are always 3 digits, with the following ranges:
+
+ * 100 to 199: resource state (started, stopped, ready, ...)
+ * 200 to 399: positive action result
+ * 400 to 599: negative action result
+ * 600 to 999: future use
+
+### List of current status codes
+
+Code  | Meaning
+---|------
+100   | Operation created
+101   | Started
+102   | Stopped
+103   | Running
+104   | Cancelling
+105   | Pending
+106   | Starting
+107   | Stopping
+108   | Aborting
+109   | Freezing
+110   | Frozen
+111   | Thawed
+200   | Success
+400   | Failure
+401   | Cancelled
+
+## Recursion
+To optimise queries of large lists, recursion is implemented for collections. A `recursion` argument can be passed to a GET query against a collection.
+
+The default value is 0 which means that collection member URLs are returned. Setting it to 1 will have those URLs be replaced by the object they point to (typically a dict).
+
+Recursion is implemented by simply replacing any pointer to a job (URL) by the object itself.
+
+## Async operations
+Any operation which take more than a second must be done in the background, returning a background operation ID to the client. With this ID, the client is able to either poll for a status update or wait for a notification using the long-poll API.
+
+## Notifications
+A web-socket based API is available for notifications. Different notification types exist to limit the traffic going to the client. It is recommended that the client always subscribes to the *operations* notification type before triggering remote operations so that it doesn't have to continually poll for their status.
+
+## PUT vs PATCH
+PUT and PATCH APIs are supported to modify existing objects.
+
+PUT replaces the entire object with a new definition, it's typically called after the current object state was retrieved through GET.
+
+To avoid race conditions, the ETag header should be read from the GET response and sent as If-Match for the PUT request. Doing so makes the request fail if the object was modified between GET and PUT.
+
+PATCH can be used to modify a single field inside an object by only specifying the property that you want to change. To unset a key, setting it to empty will usually do the trick, but there are cases where PATCH won't work and PUT needs to be used instead.
+
+## Authorisation
+Some operation may require a token to be included in the HTTP Authorisation header even if the request is already authenticated using a trusted certificate. If the token is not valid, the request is rejected by the server. This ensures that only authorised clients can access those endpoints.
+
+    `Authorization: bearer <token>`
```

Comment:
```suggestion
    Authorization: bearer <token>
```

This should already be formatted with a monospaced font as it is indented to the right.